server/api/runtime_handlers/sparkjob/spark3job.py

```diff
@@ -700,24 +693,52 @@ def _enrich_job(
             )
         return
 
-    def _enrich_node_selectors(self, runtime, job, project_name):
-        # After adding the runtime executor and driver node selectors to the job object,
+    def _enrich_node_selectors(
+        self,
+        run: mlrun.run.RunObject,
+        runtime: mlrun.runtimes.sparkjob.spark3job.Spark3Runtime,
+        job: dict,
+        project_name: str,
+    ):
+        # If `driver_node_selector` is defined, it takes precedence and will be used for the driver node selector.
+        # If not provided or empty, the `node_selector` from the run will be used instead.
```

Comment:
what is the run node selector in this context? project and config?

Reply:
Yes, but in case the user decided to do explicitly edit the function like this: `spark_fn.spec.node_selector = {"key": "val"}`, the value will be what they have defined + project and config.

Reply:
explain where the `run.spec.node_selector` was enriched from